ReSharper.FSharp/src/FSharp.Psi.Intentions/src/Intentions/FunctionAnnotationAction.fs

```diff
@@ -105,17 +107,14 @@ type FunctionAnnotationAction(dataProvider: FSharpContextActionDataProvider) =
     override x.Text = "Add type annotations"
 
     override x.IsAvailable _ =
-        let letBindings = dataProvider.GetSelectedElement<ILetBindings>()
-        if isNull letBindings then false else
-
-        let bindings = letBindings.Bindings
-        if bindings.Count <> 1 then false else
-
-        isAtLetExprKeywordOrReferencePattern dataProvider letBindings && not (isAnnotated bindings[0])
+        let binding = dataProvider.GetSelectedElement<IBinding>()
+        if isNull binding then false else
+        let letBang = LetOrUseExprNavigator.GetByBinding(binding) 
+        if isNotNull letBang && (getTokenType letBang.BindingKeyword) = FSharpTokenType.LET_BANG then false else
```

Comment:
Please also check for `and!` and `use!`.

It's sad there's no nice way to distinguish `let` expression from `let!` for now. I'll think about ways to make it simpler.